src/main/java/org/reactome/addlinks/referencecreators/TargetPathogenReferenceCreator.java

```diff
@@ -0,0 +1,219 @@
+package org.reactome.addlinks.referencecreators;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.gk.model.GKInstance;
+import org.gk.model.ReactomeJavaConstants;
+import org.gk.persistence.MySQLAdaptor;
+import org.gk.schema.InvalidAttributeException;
+
+public class TargetPathogenReferenceCreator extends SimpleReferenceCreator<String>
+{
+	private static final String REACTOME_IDENTIFIER_PATTERN_STRING = "R-[A-Z]{3}-.*";
+	private static final Pattern REACTOME_IDENTIFIER_PATTERN = Pattern.compile(REACTOME_IDENTIFIER_PATTERN_STRING);
+
+	// These sets are static so that the first instance of this class can populate them, and all other instances can
+	// just rely on the data that has already been set.
+	private static Map<String, Set<String>> reactomeToTargetPathogen = new HashMap<>();
+	private static Map<String, Set<String>> uniProtToTargetPathogen = new HashMap<>();
+
+	private long personID;
+
+	public TargetPathogenReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB)
+	{
+		super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB);
+	}
+
+	public TargetPathogenReferenceCreator(MySQLAdaptor adapter, String classToCreate, String classReferring, String referringAttribute, String sourceDB, String targetDB, String refCreatorName)
+	{
+		super(adapter, classToCreate, classReferring, referringAttribute, sourceDB, targetDB, refCreatorName);
+	}
+
+	/**
+	 * Can create references to Target Pathogen on reaction objects that are identifier by a Reactome identifier. We will also probably want to create a reference
+	 * from proteins identified by UniProt identifiers. Might be tricky to combine this functionality into one ReferenceCreator, but we'll see...
+	 */
+	@Override
+	public void createIdentifiers(long personID, Map<String, String> mapping, List<GKInstance> sourceReferences) throws Exception
+	{
+		this.personID = personID;
+		if (reactomeToTargetPathogen.size() == 0 && uniProtToTargetPathogen.size() == 0)
```

Comment:
Could it ever happen that one is populated but the other is not?

Reply:
If both sets are empty, the code between lines 53 and 86 is supposed to process ALL identifiers and populate both caches. At the end of that loop, both sets ideally should have some data. If one of them somehow does _not_ have data, then it probably means there were no identifiers for it, and trying to repopulate it separately wouldn't do much because the input will be the same as the previous attempt.